code/ai/aicode.cpp

```diff
@@ -14736,6 +14737,41 @@ static void ai_control_info_check(ai_info *aip, ship_info *sip, physics_info *pi
 		no_rot = true;
 	}
 
+	if (no_rot && no_lat) {
+		aip->ai_override_flags.reset();
+	}
+	else
+	{
+		if (aip->ai_override_flags[AI::Maneuver_Override_Flags::Dont_bank_when_turning] || sip->flags[Ship::Info_Flags::Dont_bank_when_turning])
+			AI_ci.control_flags |= CIF_DONT_BANK_WHEN_TURNING;
+		if (aip->ai_override_flags[AI::Maneuver_Override_Flags::Dont_clamp_max_velocity] || sip->flags[Ship::Info_Flags::Dont_clamp_max_velocity])
+			AI_ci.control_flags |= CIF_DONT_CLAMP_MAX_VELOCITY;
+		if (aip->ai_override_flags[AI::Maneuver_Override_Flags::Instantaneous_acceleration] || sip->flags[Ship::Info_Flags::Instantaneous_acceleration])
+			AI_ci.control_flags |= CIF_INSTANTANEOUS_ACCELERATION;
+	}
+
+	// set physics flag according to whether we are instantaneously accelerating
+	if (AI_ci.control_flags & CIF_INSTANTANEOUS_ACCELERATION)
+		pi->flags |= PF_MANEUVER_NO_DAMP;
+	else
+		pi->flags &= ~PF_MANEUVER_NO_DAMP;
+}
+
+// applies any rotational or lateral maneuver values to the AI's CI
+static void ai_control_info_apply(ai_info *aip)
+{
+	if (aip->ai_override_flags.none_set())
+		return;
+
+	bool no_lat = false;
+	bool no_rot = false;
+
+	if (!aip->ai_override_flags[AI::Maneuver_Override_Flags::Lateral_never_expire] && timestamp_elapsed(aip->ai_override_lat_timestamp))
+		no_lat = true;
+
+	if (!aip->ai_override_flags[AI::Maneuver_Override_Flags::Rotational_never_expire] && timestamp_elapsed(aip->ai_override_rot_timestamp))
+		no_rot = true;
+
 	if (no_rot && no_lat) {
 		aip->ai_override_flags.reset();
```

Comment:
Is this now redundant since all ships go through what seems to be the same check in `ai_control_info_flags_upkeep`?

Reply:
Ah, yes, that is part of upkeeping flags so there's no need for it here.